hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/util/HoodieGlueClient.java

```diff
@@ -0,0 +1,487 @@
+      }
+
+      try {
+        LinkedHashMap<String, String> mapSchema = HiveSchemaUtil.parquetSchemaToMapSchema(storageSchema, syncConfig.supportTimestamp, false);
+
+        List<Column> schemaPartitionKeys = new ArrayList<>();
+        List<Column> schemaWithoutPartitionKeys = new ArrayList<>();
+        for (String key : mapSchema.keySet()) {
+          String keyType = HiveSchemaUtil.getPartitionKeyType(mapSchema, key);
+          Column column = new Column().withName(key).withType(keyType.toLowerCase()).withComment("");
+          // In Glue, the full schema should exclude the partition keys
+          if (syncConfig.partitionFields.contains(key)) {
+            schemaPartitionKeys.add(column);
+          } else {
+            schemaWithoutPartitionKeys.add(column);
+          }
+        }
+
+        StorageDescriptor storageDescriptor = new StorageDescriptor();
+        serdeProperties.put("serialization.format", "1");
+        storageDescriptor
+            .withSerdeInfo(new SerDeInfo().withSerializationLibrary(serdeClass)
+                .withParameters(serdeProperties))
+            .withLocation(syncConfig.basePath.replaceFirst("s3a", "s3"))
+            .withInputFormat(inputFormatClass)
+            .withOutputFormat(outputFormatClass)
+            .withColumns(schemaWithoutPartitionKeys);
+
+        TableInput tableInput = new TableInput();
+        tableInput.withName(tableName)
+            .withTableType(TableType.EXTERNAL_TABLE.toString())
+            .withParameters(params)
+            .withPartitionKeys(schemaPartitionKeys)
+            .withStorageDescriptor(storageDescriptor)
+            .withLastAccessTime(new Date(System.currentTimeMillis()))
+            .withLastAnalyzedTime(new Date(System.currentTimeMillis()));
+        request.withDatabaseName(syncConfig.databaseName)
+            .withTableInput(tableInput);
+
+        CreateTableResult result = awsGlueClient.createTable(request);
+        LOG.info("Successfully created table in Glue: " + result.toString());
+      } catch (AlreadyExistsException exception) {
+        LOG.warn("Table " + tableName + " already exists in database " + syncConfig.databaseName, exception);
+      } catch (Exception exception) {
+        LOG.error("Failed to create table " + tableName + " in database " + syncConfig.databaseName, exception);
+        throw new HoodieHiveSyncException("Failed to create table " + tableName + " in database " + syncConfig.databaseName, exception);
+      }
+    }
+  }
+
+  @Override
+  public Option<String> getLastCommitTimeSynced(String tableName) {
+    try {
+      Table table = getTable(syncConfig.databaseName, tableName);
+      return Option.of(table.getParameters().getOrDefault(HOODIE_LAST_COMMIT_TIME_SYNC, null));
+    } catch (Exception e) {
+      throw new HoodieHiveSyncException("Failed to get the last commit time synced from the database", e);
+    }
+  }
+
+  @Override
+  public void updateLastCommitTimeSynced(String tableName) {
```

Comment:
can you double check and ensure caller to this method respects `--conditional-sync` flag in HiveSyncConfig?